detect/gcp/gcp.go

```diff
@@ -0,0 +1,90 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
```

Comment:
tests?

Reply:
The detector collects resources by cloud.google.com/go/compute/metadata and I think it is difficult to mock the resources since different platforms like gce and gke have different attributes. Besides, the OpenCensus auto detector did not include the tests either maybe for similar reasons.

Reply:
That is true, but tests might have helped catch that cut-paste error from earlier. I'm okay with no tests here though in the interest of expediency.